src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -679,16 +682,25 @@ private void ProcessPing(string targetNameOrAddress)
             {
                 try
                 {
-                    hostEntry = Dns.GetHostEntry(targetNameOrAddress);
+                    hostEntry = GetHostEntryWithCancel(targetNameOrAddress);
 
                     if (ResolveDestination)
                     {
                         resolvedTargetName = hostEntry.HostName;
-                        hostEntry = Dns.GetHostEntry(hostEntry.HostName);
+                        hostEntry = GetHostEntryWithCancel(hostEntry.HostName);
                     }
                 }
+                catch (PipelineStoppedException p)
+                {
+                    throw p;
```

Comment:
Didn't know that was a thing. Is there a significant difference?

Reply:
Which one you are asking about?

for the `throw`, it will keep the original stack trace; while `throw p` will rewrite the stack trace.
for the conditional catch block, it's slightly efficient because it avoids looking for a handler for the `PipelineStoppedException`. Although the efficiency doesn't matter here, it looks more concise :)

Reply:
Poked at this one for a bit and changed my mind a couple times, but it looks like the conditional catch block is about the same as just doing a general catch with an `if` statement when it comes down to it. Separate catch block seems a little tidier.

Reply:
I'm fine keeping the `catch (PipelineStoppedException p)` block, as long as you change `throw p` to `throw;`.
The main benefit of conditional exception is to avoid unnecessary search of exception handlers and the stack unwinding.